langchain_postgres/vectorstores.py

```diff
@@ -249,11 +315,13 @@ def __init__(
         engine_args: Optional[dict[str, Any]] = None,
         use_jsonb: bool = True,
         create_extension: bool = True,
+        async_mode: bool = False,
```

Comment:
@pprados  

Sync and async types occupy the same variable names:

* `connection` can be sync or async
* `session_maker` is now public and can be sync and async

I assume that `session_maker` has been made public so it can be mutated post initialization? (To support transactions?)

Would it be feasible to instead update the initializer to take in a Callable that can yield a session -- or does it not help with transactions?

```python
connection: Union[None, DBConnection, Engine, str, Callable[[], Session]],  = None,
connection_async: Union[None, async variants...],  = None,
```

The pattern would be cheap initialization of vectorstores (initialization involves 0 network requests)

---

If you think this doesn't make sense or help, then let's skip

Reply:
done